File: src/main/java/com/assets/board/service/XmlGeneratorService.java
package com.assets.board.service;

import com.assets.board.model.tax.Declar;
import com.assets.board.model.tax.DeclarBody;
import com.assets.board.model.tax.DeclarBodyF1;
import jakarta.xml.bind.JAXBContext;
import jakarta.xml.bind.Marshaller;
import org.springframework.stereotype.Service;

import java.io.File;
import java.io.StringWriter;

@Service
public class XmlGeneratorService {

    private static final String ENCODING = "windows-1251";

    public String generateDeclarXml(Declar declar) throws Exception {
        JAXBContext context = JAXBContext.newInstance(Declar.class);
        Marshaller marshaller = context.createMarshaller();

        // Configure marshaller
        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
        marshaller.setProperty(Marshaller.JAXB_ENCODING, ENCODING);
        marshaller.setProperty(Marshaller.JAXB_FRAGMENT, false);

        StringWriter writer = new StringWriter();
        marshaller.marshal(declar, writer);

        return writer.toString();
    }

    public void saveXmlToFile(Declar declar, String filepath) throws Exception {
        JAXBContext context = JAXBContext.newInstance(Declar.class, DeclarBody.class, DeclarBodyF1.class);
        Marshaller marshaller = context.createMarshaller();

        // Configure marshaller
        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
        marshaller.setProperty(Marshaller.JAXB_ENCODING, ENCODING);
        marshaller.setProperty(Marshaller.JAXB_FRAGMENT, false);


        File file = new File(filepath);
        marshaller.marshal(declar, file);
    }
}

File: src/main/java/com/assets/board/service/impl/XmlGeneratorService.kt
package com.assets.board.service.impl

import com.assets.board.model.tax.Declar
import com.assets.board.model.tax.DeclarBody
import com.assets.board.model.tax.DeclarBodyF1
import jakarta.xml.bind.JAXBContext
import jakarta.xml.bind.Marshaller
import org.springframework.stereotype.Service
import java.io.File
import java.io.StringWriter

@Service
class XmlGeneratorService {
    @Throws(Exception::class)
    fun generateDeclarXml(declar: Declar?): String {
        val context = JAXBContext.newInstance(Declar::class.java)
        val marshaller = context.createMarshaller()

        // Configure marshaller
        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true)
        marshaller.setProperty(Marshaller.JAXB_ENCODING, ENCODING)
        marshaller.setProperty(Marshaller.JAXB_FRAGMENT, false)

        val writer = StringWriter()
        marshaller.marshal(declar, writer)

        return writer.toString()
    }

    @Throws(Exception::class)
    fun saveXmlToFile(declar: Declar?, filepath: String) {
        val context = JAXBContext.newInstance(Declar::class.java, DeclarBody::class.java, DeclarBodyF1::class.java)
        val marshaller = context.createMarshaller()

        // Configure marshaller
        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true)
        marshaller.setProperty(Marshaller.JAXB_ENCODING, ENCODING)
        marshaller.setProperty(Marshaller.JAXB_FRAGMENT, false)


        val file = File(filepath)
        marshaller.marshal(declar, file)
    }

    companion object {
        private const val ENCODING = "windows-1251"
    }
}
